libsolidity/ast/Types.cpp

```diff
@@ -362,20 +381,28 @@ MemberList::MemberMap Type::boundFunctions(Type const& _type, ASTNode const& _sc
 			)
 		)
 			continue;
-		auto const& library = dynamic_cast<ContractDefinition const&>(
-			*ufd->libraryName().annotation().referencedDeclaration
-		);
-		for (FunctionDefinition const* function: library.definedFunctions())
+
```

Comment:
Here I also wonder if checking ``ufd->usesBraces()`` and asserting accordingly would be more fool-proof.
But yeah - on the other hand "usesBraces" is a syntactic property and it makes sense not to consider it anymore later on - resp. again: also fine as is.